### PR TITLE
test(core): Add unit tests for has-facet-values promotion condition

### PR DESCRIPTION
**Pull request by GabrielRoc**
## Description

Adds a direct unit test suite for the `at_least_n_with_facets` promotion condition (`has-facet-values-condition.ts`), which previously had **0% function coverage**.

The condition delegates per-line facet matching to `FacetValueChecker` and sums the qualifying quantities against `minimum`. Because `FacetValueChecker` is a NestJS-injected service, the test injects a `vi.fn()` mock via the condition's `init()` hook — no database or TestServer is required.

7 tests covering:
- **Boundary-value analysis** on total matching quantity vs `minimum` (below / at / above)
- Correct aggregation across multiple lines
- Correct exclusion of non-matching lines
- Verification that `ctx` and the facet ID list are forwarded as-is to `hasFacetValues()`

## Breaking changes

None.

## Checklist

- [x] I have set a clear title
- [x] My PR is small and contains a single feature
- [x] I have added or updated test cases

Relates to #4835

---
<a href="https://app.blacksmith.sh/vendurehq/codesmith/vendure/pr/4882"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-dark-v2.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-light-v2.svg">View with Codesmith</picture></a> <a href="https://backend.blacksmith.sh/track/enable-autofix?expires=1785127873&installation_id=128408429&pr_number=4882&repository=vendurehq%2Fvendure&return_to=https%3A%2F%2Fgithub.com%2Fvendurehq%2Fvendure%2Fpull%2F4882&signature=59284c815d96171f6531b7eea0ea83b89c2e30238ea94dbb2371d8a88b1764c3"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-light.svg">Autofix with Codesmith</picture></a>
<sup>Need help on this PR? Tag <code>/codesmith</code> with what you need. Autofix is disabled.</sup>

### CODE REVIEW

**Comment by GabrielRoc** [comment]
I have read the CLA Document and I hereby sign the CLA

File: packages/core/src/config/promotion/conditions/has-facet-values-condition.spec.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+function orderWith(lines: OrderLine[]): Order {
+    return new Order({ lines });
+}
+
+const ctx = createRequestContext({ pricesIncludeTax: false });
+
+async function check(order: Order, args: ConfigArg[]) {
+    return hasFacetValues.check(ctx, order, args, undefined as any);
+}
```
[the remark this answers is not in the document]
Comment (GabrielRoc) [reply packages/core/src/config/promotion/conditions/has-facet-values-condition.spec.ts:62]: Dismissed. `hasFacetValues.check()` does not use the `promotion` argument. Passing `undefined as any` is the intentional, consistent pattern across all promotion-condition specs in this PR series.

File: packages/core/src/config/promotion/conditions/has-facet-values-condition.spec.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    describe('checker invocation', () => {
+        it('passes the ctx and facet ID list to hasFacetValues', async () => {
+            const order = orderWith([line('v1', 1)]);
+            const args = buildArgs(1, ['f1', 'f2']);
+            mockHasFacetValues.mockResolvedValue(true);
+
+            await check(order, args);
+
+            expect(mockHasFacetValues).toHaveBeenCalledWith(order.lines[0], ['f1', 'f2'], ctx);
+        });
```
[the remark this answers is not in the document]
Comment (GabrielRoc) [reply packages/core/src/config/promotion/conditions/has-facet-values-condition.spec.ts:136]: Valid point. Fixed in commit `8da9b635c`: added `mockHasFacetValues.mockClear()` at the start of the test to reset the call history, and added `expect(mockHasFacetValues).toHaveBeenCalledTimes(1)` to pin the assertion to exactly one invocation from this test. The `toHaveBeenCalledWith` assertion is now unambiguous even if more tests are added later.

**Review by grolmus — APPROVED** [review]
Excellent, well-structured coverage — boundary-value analysis on the thresholds, clean injector-based mocking (no DB), and assertions that the condition is invoked with the right args. Test-only, no production changes. Approving.
